### PR TITLE
Implement `Plutip.Spawn`, update ODC &amp; use its CLI args in plutip env

### LINKED ISSUE #616: Plutip: ODC: use CLI options instead of config
Following this change, we should change how we launch ODC:

https://github.com/Plutonomicon/cardano-transaction-lib/pull/589

### PR DESCRIPTION
**Pull request by klntsky**
Closes #616 

### Pre-review checklist

- [x] All code has been formatted using our config (`make format` for Purescript, `nixpkgs-fmt` for Nix)
- [x] All Purescript imports are explicit, including constructors
- [ ] **All** existing examples have been run locally against a fully-synced testnet node
- [x] The integration and unit tests have been run (`npm run test`) locally
- [ ] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`)
instead of config

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
     # for the purescript project
     ogmios.url = "github:mlabs-haskell/ogmios";
-    ogmios-datum-cache.url = "github:mlabs-haskell/ogmios-datum-cache/1be5e903e14f5a1785983bdb38203195b401bd15";
+    ogmios-datum-cache.url = "github:mlabs-haskell/ogmios-datum-cache/85e030ae914abcffba4a7c261d9cd81f98b2168c";
```
Comment (Luis-omega) [inline flake.nix:12]:
```suggestion
    ogmios-datum-cache.url = "github:mlabs-haskell/ogmios-datum-cache/def286f8e0e3c54707ef69578ac71a5e81c708a9";
```
This has control of the `logging` level.

**Review by marijanp — APPROVED** [review]
LGTM, please resolve the other comments first.
It would have been nice to see some tests for `Plutip.Spawn`, however I ran the `Plutip.Tests` manually and everything passed.

File: src/Plutip/Spawn.purs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  child <- spawn cmd args opts
+  onExit child $ const $ cont $ Left $ error $ "Process " <> cmd <> " exited"
+  ref <- Ref.new ""
+  onDataString (stdout child) Encoding.UTF8
+    \str -> do
+      Ref.modify_ (_ <> str) ref
+      output <- Ref.read ref
+      case filter output of
+        NoOp -> pure unit
+        Success -> cont (pure child)
```
Comment (firefrorefiddle) [inline src/Plutip/Spawn.purs:63]: Will this continue to listen for events as long as the child lives?

Comment (klntsky) [reply src/Plutip/Spawn.purs:63]: You are right, this is inacceptable.

**Review by firefrorefiddle — COMMENTED** [review]
LGTM

**Pull request merged**